ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/admin/ApiPartyManagementService.scala

```diff
@@ -52,58 +58,50 @@ class ApiPartyManagementService private (
-    val newParty = result.partyDetails.get.party
-    val description = s"party $newParty"
-
-    PollingUtils
-      .pollUntilPersisted(partyManagementService.listParties _)(
-        _.exists(_.party == newParty),
-        description,
-        50.milliseconds,
-        500.milliseconds,
-        d => d * 2,
-        scheduler,
-        loggerFactory)
-      .map { numberOfAttempts =>
-        logger.debug(s"Party $newParty available, read after $numberOfAttempts attempt(s)")
-        result
-      }(DE)
-  }
-
   override def allocateParty(request: AllocatePartyRequest): Future[AllocatePartyResponse] = {
     val party = if (request.partyIdHint.isEmpty) None else Some(request.partyIdHint)
     val displayName = if (request.displayName.isEmpty) None else Some(request.displayName)
+    val submissionId = Ref.LedgerString.assertFromString(UUID.randomUUID().toString)
 
     FutureConverters
-      .toScala(writeService
-        .allocateParty(party, displayName))
+      .toScala(
+        writeService
+          .allocateParty(party, displayName, submissionId))
       .flatMap {
-        case PartyAllocationResult.Ok(details) =>
-          Future.successful(AllocatePartyResponse(Some(mapPartyDetails(details))))
-        case r @ PartyAllocationResult.Overloaded =>
+        case SubmissionResult.Acknowledged =>
+          pollForAllocationResult(submissionId).flatMap {
+            case domain.PartyAllocationEntry.Accepted(_, _, partyDetails) =>
+              Future.successful(AllocatePartyResponse(Some(
+                PartyDetails(partyDetails.party, partyDetails.displayName.getOrElse(""), true))))
+            case domain.PartyAllocationEntry.Rejected(_, _, reason) =>
+              Future.failed(ErrorFactories.invalidArgument(reason))
+          }
+        case r @ SubmissionResult.Overloaded =>
           Future.failed(ErrorFactories.resourceExhausted(r.description))
-        case r @ PartyAllocationResult.AlreadyExists =>
-          Future.failed(ErrorFactories.invalidArgument(r.description))
-        case r @ PartyAllocationResult.InternalError(_) =>
+        case r @ SubmissionResult.InternalError(_) =>
           Future.failed(ErrorFactories.internal(r.reason))
-        case r @ PartyAllocationResult.InvalidName(_) =>
-          Future.failed(ErrorFactories.invalidArgument(r.description))
-        case r @ PartyAllocationResult.ParticipantNotAuthorized =>
-          Future.failed(ErrorFactories.permissionDenied(r.description))
-        case r @ PartyAllocationResult.NotSupported =>
+        case r @ SubmissionResult.NotSupported =>
           Future.failed(ErrorFactories.unimplemented(r.description))
       }(DE)
-      .flatMap(pollUntilPersisted)(DE)
+  }
+
+  private def pollForAllocationResult(
+      submissionId: SubmissionId): Future[domain.PartyAllocationEntry] = {
+    PollingUtils
+      .pollSingleUntilPersisted(() =>
+        partyManagementService.lookupPartyAllocationEntry(submissionId))(
```

Comment:
that stream was the initial idea to be able to return all but it ended up being more than was needed since, at the moment, really we only need single lookup by submissionId so i will probably kill getPartyAllocationEntries (preferably as a follow-on PR)

Reply:
i've killed getPartyAllocationEntries and partyAllocationEntries on this PR now